web/source/osk/oskManager.ts

```diff
@@ -231,20 +249,100 @@ namespace com.keyman.osk {
       }
     }
 
-    private get width(): number {
-      if(this.vkbd) {
-        return this.vkbd.width;
-      } else {
-        return this._baseWidth;
+    /**
+     * The configured width for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic width scaling. 
+     */
+     get width(): LengthStyle {
+      return this._width;
+    }
+
+    /**
+     * The configured height for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic height scaling. 
+     */
+    get height(): LengthStyle {
+      return this._height;
+    }
+
+    /**
+     * The computed width for this VisualKeyboard.  May be null if auto sizing
+     * is allowed and the VisualKeyboard is not currently in the DOM hierarchy.
+     */
+    get computedWidth(): number {
+      // if(!this.kbdDiv) {
+      //   // Intermediate state - can be called during VisualKeyboard's constructor, before
+      //   // computedHeight can receive a value.
+      //   return undefined;
+      // }
+
+      // Computed during layout operations; allows caching instead of continuous recomputation.
+      if(this.needsLayout) {
+        this.refreshLayout();
+      }
+      return this._computedWidth;
+    }
+
+    /**
+     * The computed height for this VisualKeyboard.  May be null if auto sizing
+     * is allowed and the VisualKeyboard is not currently in the DOM hierarchy.
+     */
+    get computedHeight(): number {
+      // if(!this.kbdDiv) {
+      //   // Intermediate state - can be called during VisualKeyboard's constructor, before
+      //   // computedHeight can receive a value.
+      //   return undefined;
+      // }
+
+      // Computed during layout operations; allows caching instead of continuous recomputation.
+      if(this.needsLayout) {
+        this.refreshLayout();
+      }
+      return this._computedHeight;
+    }
+
+    /**
+     * The top-level style string for the font size used by the predictive banner 
+     * and the primary keyboard visualization elements.
+     */
+    get baseFontSize(): string {
+      return this.parsedBaseFontSize.styleString;
```

Comment:
Is `this.parsedBaseFontSize` always going to be available? It doesn't seem to be set in the constructor.

Reply:
... that's what the getter that directly follows this block is for.

Reply:
Yeah, but could `baseFontSize` ever be called by another component before the getter below is? Just in terms of future robustness.

Reply:
I don't see the issue here.  This _directly calls the getter_, which is written to perform lazy-initialization if/when necessary.  If there's a part of that getter that isn't robust, please point it out to me.

Reply:
oh duh. I don't know why it didn't click that the `get` was for this...!